src/daemon/daemon.cpp

```diff
@@ -1712,36 +1725,74 @@ try // clang-format on
         return grpc::Status::OK;
     };
 
-    // TODO@snapshots retrieve snapshot names to gather info
+    std::unordered_map<std::string, std::unordered_set<std::string>> instance_snapshots_map;
     auto fetch_snapshot_overview = [&](VirtualMachine& vm) {
         const auto& name = vm.vm_name;
-        auto overview = response.mutable_snapshot_overview()->add_overview();
-        auto fundamentals = overview->mutable_fundamentals();
 
-        overview->set_instance_name(name);
-        fundamentals->set_snapshot_name("snapshot1");
-        fundamentals->set_comment("This is a sample comment");
+        auto get_snapshot_info = [&](std::shared_ptr<const Snapshot> snapshot) {
+            auto overview = response.mutable_snapshot_overview()->add_overview();
+            auto fundamentals = overview->mutable_fundamentals();
+
+            overview->set_instance_name(name);
+            fundamentals->set_snapshot_name(snapshot->get_name());
+            fundamentals->set_parent(snapshot->get_parent_name());
+            fundamentals->set_comment(snapshot->get_comment());
+            // TODO@snapshots populate snapshot creation time once available
+        };
+
+        if (const auto& it = instance_snapshots_map.find(name);
+            it == instance_snapshots_map.end() || it->second.empty())
+        {
+            for (const auto& snapshot : vm.view_snapshots())
+                get_snapshot_info(snapshot);
+        }
+        else
+        {
+            for (const auto& snapshot : it->second)
+            {
+                try
+                {
+                    get_snapshot_info(vm.get_snapshot(snapshot));
+                }
+                catch (const std::out_of_range&)
+                {
+                    fmt::memory_buffer errors;
+                    add_fmt_to(errors, "snapshot \"{}\" does not exist", snapshot);
+
+                    return grpc_status_for(errors, grpc::StatusCode::NOT_FOUND);
+                }
+            }
+        }
 
         return grpc::Status::OK;
     };
 
-    mp::InstanceNames instance_names;
-    for (const auto& n : request->instances_snapshots())
-        instance_names.add_instance_name(n.instance_name());
-
     auto [instance_selection, status] =
-        select_instances_and_react(operative_instances, deleted_instances, instance_names.instance_name(),
+        select_instances_and_react(operative_instances, deleted_instances, request->instances_snapshots(),
```

Comment:
Note that regular `info` will also need to handle snapshots. Currently we get:

```
$ mp info t.good
Name:           t
State:          Stopped
Snapshots:      4
IPv4:           --
Release:        --
Image hash:     afb820a92602 (Ubuntu 22.04 LTS)
CPU(s):         --
Load:           --
Disk usage:     --
Memory usage:   --
Mounts:         --
```

We can leave that to snapshot details, but a TODO in the code wouldn't hurt.